source/gateway.go

```diff
@@ -307,7 +307,13 @@ func (c *gatewayRouteResolver) resolve(rt gatewayRoute) (map[string]endpoint.Tar
 	for _, rps := range rt.RouteStatus().Parents {
 		// Confirm the Parent is the standard Gateway kind.
 		ref := rps.ParentRef
+
 		namespace := strVal((*string)(ref.Namespace), meta.Namespace)
+		if rps.Conditions[0].ObservedGeneration != meta.Generation {
```

Comment:
what if len(rps.Conditions) == 0?

Reply:
If conditions are of length 0, then the gateway controller hasn't accepted the route at all and external-dns should not act on it at all. Yes we will need a check here for that, I migrated this to the accepted check as we also need to make sure the route is accepted and is of the current generation.